src/transformers/generation/utils.py

```diff
@@ -1751,7 +1758,8 @@ def generate(
         )
 
         use_dynamic_cache_by_default = False
-        if generation_config.cache_implementation is not None and model_kwargs.get("past_key_values") is not None:
+        cache_name = getattr(self, "cache_name", "past_key_values")
```

Comment:
might be better to set this as a class attribute, all that inherit from `Cache` will have `"path_key_values"` and mamba will get `"cache_params"` WDYT?

Reply:
currently `MambaCache` is not inherited from `Cache` because of APIs of `Cache` are only suitable for transformer models with kv states, so you mean make `cache_name` a class attribute of `Cache` and `MambaCache` with values being `past_key_values` and `cache_params`?

Reply:
`cache_name = "cache_params"` for mamba cache class, and `cache_name = "past_key_values"` for `Cache` classes !

Reply:
I also prefer a more verbose version for now. I spent some time looking for the `cache_name` variable in this review, which is not a good indicator of readability :D

e.g. 

```py
if "mamba" in self.__class__.__name__.lower():
    cache_var_name = "cache_params"
else:
    cache_var_name = "past_key_values"
```

Reply:
Ok, I guess it's a way to see if we are using mamba-related models, and there is an issue with associating `cache_name` with `Cache`, we need to know which cache we are creating in order to know the cache name, which brings a circular issue when we are trying to check if users are passing both `cache_implementation` and a cache instance, let's go with it for now.